src/behavior.cpp

```diff
@@ -0,0 +1,90 @@
+#include "behavior.h"
+
+using namespace behavior;
+
+namespace behavior
+{
+sequential default_sequential;
+fallback default_fallback;
+}
+
+std::pair<status, const node *> sequential::evaluate( const Creature *subject,
+        const std::vector<const node *> children ) const
+{
+    for( const node *child : children ) {
+        std::pair<status, const node *> outcome = child->tick( subject );
+        switch( outcome.first ) {
+            case running:
+            case failure:
+                return outcome;
+            case success:
+                ;
```

Comment:
This is kind of ugly, but every other way I can think to write it just moves the ugly around.  I'm not happy about it but I can't offer a better solution.

Reply:
IMO there's nothing wrong with:
```c++
if( outcome.first == running || outcome.first == failure ) { ...
```

Reply:
Given the expectation that it will only be three statuses, yea that's cleaner.